lib/Doctrine/Connection/Sqlite.php

```diff
@@ -84,11 +90,21 @@ public function __construct(Doctrine_Manager $manager, $adapter)
     public function connect() 
     {
         if ($this->isConnected) {
-            return false;
+            return;
         }
 
+        // If customer configure it
+        $hasConfigureStringify = (isset($this->pendingAttributes[Doctrine_Core::ATTR_STRINGIFY_FETCHES]));
+
         parent::connect();
 
+        if(!$hasConfigureStringify) {
+            // PHP8.1 require default to true to keep BC
+            // https://www.php.net/manual/en/migration81.incompatible.php#migration81.incompatible.pdo.sqlite
+            // Can be overwritten by user later
+            $this->dbh->setAttribute(PDO::ATTR_STRINGIFY_FETCHES, true);
```

Comment:
In the case if the user modify dbh attribute with PDO::ATTR_STRINGIFY_FETCHES, he do not respect doctrine1 behavior, it will be overwritten.

If the user set Doctrine_Core::ATTR_STRINGIFY_FETCHES to false, the value will still be false, and the new php 8.1 behavior will apply.

Doctrine3 decided to drop the handling, delegating the fault to the pdo driver. This is normal for an evolutive framework to move to the new behavior
https://github.com/doctrine/dbal/pull/5229/files

As doctrine1 target is to keep max BC, it's better to do our max to keep the previous behavior